install/installer/README.md

```diff
@@ -13,3 +13,38 @@ The best way to get started with Gitpod is by using our recommended & default in
 > The installer is an internal tool and as such not expected to be used by those external to Gitpod. Instructions for how to use it can be found in the [docs](https://github.com/gitpod-io/gitpod/tree/main/install/installer/docs/overview.md) folder of this repo.
 
 [![Gitpod ready-to-code](https://img.shields.io/badge/Gitpod-ready--to--code-908a85?logo=gitpod)](https://gitpod.io/from-referrer/)
+
+## Configuration expectations
+
+The [config](./pkg/config/) is the primary way that the installer can be interacted with. This guide details the expectations that users can have of the installer's configuration and the contract that it specifies.
+
+### Versioning
+
+The installer supports only one active version of the configuration at a given time.
+
+#### Expectations for a version
+
+The configuration for a version has the following expectations. If a constraint needs to be broken to support a new feature, this should trigger a new version of the configuration surface.
+
+1. new parameters may be introduced
+2. a parameter may have additional validation rules introduced
+3. no parameter may be removed or renamed
```

Comment:
If you remove a config parameter from the struct (eg, `Domain`), that's an absolute breaking change. If it's not then there's almost nothing that could cause a breaking change

Reply:
Hmm, I would like to understand the reason for this opinion. You've repeated that but not given a reason for your opinion. 

In my opinion, a breaking change is when the user needs to take manual action when they update the installer to a new version. When we decide that we don't need the domain config value for whatever reason anymore, we can just ignore that value when reading the config file and continue upgrading to the new Gitpod version. The user does not necessarily need to remove it from the config file for that. Thus, no manual action is needed.

In contrast, when we add a new config value that is mandatory but has no sensible default value the user needs to take action. Otherwise, the upgrade fails. That's what I would call a breaking change regarding the config contract. And it's definitely a breaking change when we rename a value or change the structure/type of a value (string type to object type etc.).